common/flatpak-transaction-log.c

```diff
@@ -0,0 +1,305 @@
+/* flatpak-transaction-log.c
+ *
+ * Copyright (C) 2017 Endless Mobile, Inc
+ *
+ * This file is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Sam Spilsbury <sam@endlessm.com>
+ */
+
+#include "config.h"
+
+#include <json-glib/json-glib.h>
+
+#include "flatpak-transaction-log.h"
+
+struct FlatpakTransactionLog
+{
+  GObject    parent;
+
+  GFile      *path;
+};
+
+typedef struct
+{
+  GObjectClass parent_class;
+} FlatpakTransactionLogClass;
+
+G_DEFINE_TYPE (FlatpakTransactionLog, flatpak_transaction_log, G_TYPE_OBJECT);
+
+enum {
+  PROP_0,
+  PROP_PATH,
```

Comment:
Nitpick: Drop `PROP_0` and just set `PROP_PATH = 1` to save defining an identifier which should never be used.